Source/Core/Editor/ProcessAssets/ProcessAssetManager.cs

```diff
@@ -99,41 +102,88 @@ internal static void Save(IProcess process)
+                    filesToDelete.Remove(filesToDelete.FirstOrDefault(file => file.EndsWith(fullFileName)));
+                    string path = $"{ProcessAssetUtils.GetProcessAssetDirectory(process.Data.Name)}/{fullFileName}";
 
-                byte[] processData = EditorConfigurator.Instance.Serializer.ProcessToByteArray(process);
+                    WriteFileIfChanged(assetData[fileName], path);
+                }
 
-                if(Enumerable.SequenceEqual(storedData, processData) == false)
+                if (EditorConfigurator.Instance.ProcessAssetStrategy.CreateManifest)
                 {
-                    WriteProcess(path, processData);
-                    Debug.Log($"Process saved to \"{path}\"");
+                    byte[] manifestData = CreateSerializedManifest(assetData);
+                    string fullManifestName = $"{BaseRuntimeConfiguration.ManifestFileName}.{EditorConfigurator.Instance.Serializer.FileFormat}";
+                    string manifestPath = $"{ProcessAssetUtils.GetProcessAssetDirectory(process.Data.Name)}/{fullManifestName}";
+
+                    WriteFileIfChanged(manifestData, manifestPath);
+
+                    filesToDelete.Remove(filesToDelete.FirstOrDefault(file => file.EndsWith($"{fullManifestName}")));
                 }
+
+                DeleteFiles(filesToDelete);
             }
             catch (Exception ex)
             {
                 Debug.LogError(ex);
             }
         }
 
-        private static void WriteProcess(string path, byte[] processData)
+        private static void DeleteFiles(IEnumerable<string> filesToDelete)
+        {
+            foreach (string file in filesToDelete)
+            {
+                Debug.Log($"File deleted: {file}");
+                File.Delete(file);
+            }
+        }
+
+        private static byte[] CreateSerializedManifest(IDictionary<string, byte[]> assetData)
+        {
+            IProcessAssetManifest manifest = new ProcessAssetManifest()
+            {
+                AssetStrategyTypeName = EditorConfigurator.Instance.ProcessAssetStrategy.GetType().FullName,
+                AdditionalFileNames = assetData.Keys.Where(name => name != assetData.Keys.First()).ToList(),
+                ProcessFileName = assetData.Keys.First(),
+            };
+
+            byte[] manifestData = EditorConfigurator.Instance.Serializer.ManifestToByteArray(manifest);
+            return manifestData;
+        }
+
+        private static void WriteFileIfChanged(byte[] data, string path)
+        {
+            byte[] storedData = new byte[0];
+
+            if (File.Exists(path))
+            {
+                storedData = File.ReadAllBytes(path);
+            }
+
+            if (Enumerable.SequenceEqual(storedData, data) == false)
+            {
+                AssetDatabase.MakeEditable(path);
+                WriteProcessFile(path, data);
+                Debug.Log($"File saved: \"{path}\"");
```

Comment:
I'm not sure if its really needed if the file is locked WriteProcessFile probably throw's the exception then.

```suggestion
                if (AssetDatabase.MakeEditable(path))
                {
                    WriteProcessFile(path, data);
                    Debug.Log($"File saved: \"{path}\"");
                }
                else
                {
                    Debug.LogError($"Saving of \"{path}\" failed! Could not make it editable.");
                }
```